ts/screens/services/ServicesHomeScreen.tsx

```diff
@@ -406,34 +667,91 @@ class ServicesHomeScreen extends React.Component<Props, State> {
 }
 
 const mapStateToProps = (state: GlobalState) => {
-  const { services } = state.entities;
-
   const potUserMetadata = userMetadataSelector(state);
   // TODO: disable selection of areas of interest if the user metadata are not loaded
   // (it causes the new selection is not loaded) https://www.pivotaltracker.com/story/show/168312476
   const userMetadata = pot.getOrElse(potUserMetadata, undefined);
 
-  const isAnyServiceLoading =
-    Object.keys(services.byId).find(k => {
-      const oneService = services.byId[k];
-      return oneService !== undefined && pot.isLoading(oneService);
-    }) !== undefined;
+  const localServicesSections = localServicesSectionsSelector(state);
+  const selectableOrganizations = localServicesSections.map(
+    (section: ServicesSectionState) => {
+      return {
+        name: section.organizationName,
+        fiscalCode: section.organizationFiscalCode
+      };
+    }
+  );
 
-  const isLoading =
-    pot.isLoading(state.entities.services.visible) ||
-    isAnyServiceLoading ||
-    pot.isLoading(potUserMetadata);
+  const localSections = selectedLocalServicesSectionsSelector(state);
+  const nationalSections = nationalServicesSectionsSelector(state);
+  const allServicesSections = notSelectedServicesSectionsSelector(state);
+
+  const getTabSevicesId = (
```

Comment:
If you want avoid object mutation you could use **_reduce_**

```typescript
const getTabSevicesId = (
    tabServices: ReadonlyArray<ServicesSectionState>
  ) => {
    return tabServices.reduce(
      (acc: ReadonlyArray<string>, curr: ServicesSectionState) => {
        const servicesOfSection = curr.data.reduce(
          (
            acc2: ReadonlyArray<string>,
            curr2: pot.Pot<ServicePublic, Error>
          ) => {
            if (pot.isSome(curr2)) {
              return [...acc2, curr2.value.service_id];
            }
            return acc2;
          },
          []
        );
        return [...acc, ...servicesOfSection];
      },
      []
    );
  };
```

Reply:
I updated it with the proposed approach within https://github.com/teamdigitale/io-app/pull/1269/commits/383772845fa1b1dd59a4d61d5512ba2520b8f0ce. It allows us to avoid object mutation. 

Anyway, the usage of `reduce` seems to me more expensive. We have no significant time usage if compared to the current number of services, but it could be discussed again when the focus will be on increase the app efficiency.

Reply:
@valentinacala Yes I would like to talk more about what is the better solution for these cases. Also for me this is no the "silver bullet" so it could be improved